internal/adapters/runtime/kubernetes/game_room.go

```diff
@@ -84,3 +90,25 @@ func (k *kubernetes) CreateGameRoomName(ctx context.Context, scheduler entities.
 	}
 	return fmt.Sprintf("%s-%s", base, utilrand.String(randomLength)), nil
 }
+
+func (k *kubernetes) createKubernetesEvent(ctx context.Context, schedulerID string, gameRoomID string, reason string, message string) error {
+	pod, err := k.clientSet.CoreV1().Pods(schedulerID).Get(ctx, gameRoomID, metav1.GetOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return errors.NewErrNotFound("game room '%s' not found", gameRoomID)
+		}
+
+		return errors.NewErrUnexpected("error getting game room instance: %s", err)
+	}
+
+	k.getEventRecorder(pod).Event(pod, corev1.EventTypeNormal, reason, message)
+
+	return nil
+}
+
+func (k *kubernetes) getEventRecorder(pod *corev1.Pod) record.EventRecorder {
+	eventBroadcaster := record.NewBroadcaster()
```

Comment:
Do we need to create a new broadcaster + start recording each time we send an event? Couldn't we instantiate this when creating the `kubernetes` struct and just access via a getter (create if `nil` or return the instance)?